src/Pmi/Application/HpoApplication.php

```diff
@@ -416,6 +416,19 @@ protected function beforeCallback(Request $request, AbstractApplication $app)
                 return $this->forwardToRoute('selectSite', $request);
             }
         }
+
+        // Display cross-awardee warning on all participant pages
+        if ($request->attributes->has('participantId')) {
+            $participantId = $request->attributes->get('participantId');
+            // Check session value
+            if (empty($this['session']->get('agreeCrossOrg_' . $participantId))) {
+                $participant = $this['pmi.drc.participants']->getById($participantId);
```

Comment:
Just wanted to confirm... the extra request here to retrieve the participant from the RDR shouldn't have any noticeable performance impact, because even though the same request will be made later on, it should be cached, right?

Reply:
Yes, if the cache is enabled (`rdr_disable_cache: 0`) it won't make another request.